src/BloomExe/NavigationIsolator.cs

```diff
@@ -148,11 +148,21 @@ private void BrowserOnDocumentCompleted(object sender, EventArgs eventArgs)
 		private void ForceDocumentCompleted()
 		{
 			Cleanup();
-			if (_pending.Count > 0)
+			while (_pending.Count > 0)
 			{
 				var task = _pending[0];
 				_pending.RemoveAt(0);
+				// Somehow raw files in /tmp can disappear after being sent to NavigationIsolator.Navigate() and before
+				// being displayed via the pending mechanism.  If the file doesn't exist, calling _current.Browser.Navigate()
+				// results in the program silently stopping on Linux.  (Going "poof" is how I would describe it.)
+				// See https://jira.sil.org/browse/BL-863 for a bug report.
+				if (task.Url.StartsWith("/tmp/") && !System.IO.File.Exists(task.Url))
+				{
+					Debug.WriteLine(String.Format("DEBUG: NavigationIsolator.ForceDocumentCompleted(): new file to display (\"{0}\") does not exist!??", task.Url));
```

Comment:
We could but first we have to find out why that file is being fed to this. Should be easy I guess, since you can reproduce the bug, just stop and look at the call stack. Also, it would be really good to know why this is disappearing on Linux; Bloom relies pretty heavily on tmp files, so we can't afford to have an unknown in this area. Race conditions are normally so hard to reproduce, but if I understand correctly, you are able to reproduce this one at will. I hope we can take advantage of that to get to the bottom of this.